tamr_toolbox/dataset/_profile.py

```diff
@@ -0,0 +1,57 @@
+"""
+Additional functions to manipulate the profile of the dataset.
+"""
+
+from tamr_unify_client.dataset.resource import Dataset, DatasetProfile
+from requests import HTTPError
+import logging
+
+LOGGER = logging.getLogger(__name__)
+
+
+def get_profile(dataset: Dataset, allow_create_or_refresh: bool = False) -> DatasetProfile:
+    """
+    Returns a dataset profile object.  Optionally can refresh or create profile if missing or
+    out-of-date.
+    Args:
+        dataset: Tamr dataset object
+        allow_create_or_refresh: optional bool to allow creation/refreshing of profile info
+    Returns:
+        DatasetProfile object
+        Warning if profile information is out of date and allow_create_or_refresh is False
+    Raises:
+        RuntimeError: if profile has not been created and allow_create_or_refresh is False
+    """
+    try:
+        profile = dataset.profile()
+    except HTTPError as e:
+        # profile not yet created or in inconsistent state: 409 or 400 return respectively
+        if e.response.status_code in [409, 400]:
```

Comment:
I am not sure what the 400 comes from. 409 is if the profile has not been created and 404 is if the dataset does not exist (which this function cannot address). If the profile is out of date, it still gets returned without any error or warnings prior to this function. Perhaps @olivito might know since I based this off a version he provided a link to.

Reply:
The comments mention 400 as an "inconsistent state". I am not sure what exactly this refers to.

Reply:
I believe I got 400 from the profile being in a bad state after a canceled job.  At least that was my commit message at the time:
https://github.com/Datatamer/customers-abbvie-clinical/commit/c57a1f6d681a0df7596eaab8b7683c296ed5f0c6

Reply:
I suppose that is then resolved by rerunning the get profile job, thanks for clarifying